### DISCUSSION TITLE
Error when attempting to regrid ORCA tripolar grid to rectilinear grid

### DISCUSSION THREAD

**Discussion by thomaswilder** [SciTools/iris-esmf-regrid#323] [Q&A]
I am attempting to regrid data from an orca025 grid to a rectilinear grid so the orca data can be compared with climatology data e.g. EN.4.

I am hoping someone can help me with an error I am receiving when running `rg = MeshToGridESMFRegridder(meshcube, target_cube)`  where the error is 

```
File "/home/users/twilder/Python/ORCA025/regrid_iris/lib/python3.10/site-packages/esmf_regrid/experimental/unstructured_regrid.py", line 104, in _as_esmf_info
    elemType = self.fnc.count(axis=1)
AttributeError: 'numpy.ndarray' object has no attribute 'count'. Did you mean: 'conj'?
```
The error file is [14520233.err.txt](https://github.com/SciTools-incubator/iris-esmf-regrid/files/13673702/14520233.err.txt)

After following [this example](https://scitools-iris.readthedocs.io/en/stable/further_topics/ugrid/other_meshes.html#nemo-data-on-orca-tripolar-grid) on how to create an orca meshcube, I have acquired the following output

```
sea_water_conservative_temperature / (degC) (-- : 1740494)
    Mesh coordinates:
        latitude                                x
        longitude                               x
    Mesh:
        name                                unknown
        location                            face
    Cell methods:
        0                                   time: mean (thickness weighted)


    Attributes:
        Conventions                         'CF-1.6'
```
which is similar to the example in the above link.

My target cube takes the form

```
conservative_temperature / (degC)   (latitude: 173; longitude: 360)
    Dimension coordinates:
        latitude                             x               -
        longitude                            -               x
    Scalar coordinates:
        depth                       0.5057600140571594 m
```

Could anyone advise why I am receiving this error, and how i might fix it?

Thanks in advance

**Comment by trexfeathers** [comment]
Thanks for raising this, @thomaswilder! And thanks for making use of the ORCA conversion example.

Looks like the function in question needs `self.fnc` - the face-node connectivity - to always be a masked array:

https://github.com/SciTools-incubator/iris-esmf-regrid/blob/d1f1c8ffb83107fb223688f3e059c624f274d07c/esmf_regrid/experimental/unstructured_regrid.py#L42-L45

And it looks like only masked arrays have the `.count()` method.

However [the Iris docs](https://scitools-iris.readthedocs.io/en/stable/generated/api/iris.experimental.ugrid.mesh.html#iris.experimental.ugrid.mesh.Connectivity) state that a `Connectivity` made from a non-masked array is totally legitimate, and that is what is done in the ORCA conversion example.

So iris-esmf-regrid needs to allow for a connectivity that is a non-masked array. The operations below are the only ones that happen to `self.fnc`, so at first glance we just need an alternative to using `.count()` and everything will be fine. @stephenworsley have I missed anything?

https://github.com/SciTools-incubator/iris-esmf-regrid/blob/d1f1c8ffb83107fb223688f3e059c624f274d07c/esmf_regrid/experimental/unstructured_regrid.py#L99-L106

**Reply by stephenworsley** [reply]
@trexfeathers Yes, this seems to be what's going on. I expect that we could probably add some logic here that handles unmasked arrays separately.

**Comment by trexfeathers** [comment]
If I'm correct then it would be appropriate to convert this from a discussion to an issue.

**Reply by thomaswilder** [reply]
Thanks @trexfeathers, is there a quick fix for this then? Would this meshcube need masking with say the nemo tmask?

**Reply by trexfeathers** [reply]
The quick fix would be to just make sure `face_node_inds` (from the docs example) is a masked array using `numpy.ma`. None of the points need to be masked, so it will work identically as before.

**Reply by thomaswilder** [reply]
I can  confirm that this is now working, with orca025 regridding to the lat-lon 1 deg grid. I am happy to create an issue?

**Reply by trexfeathers** [reply]
#329 👍

**Comment by ESadek-MO** [comment]
Closing this, as it's been reported in #329
